### PR TITLE
feat/PN-12889: custom language detector in PF/PG/PA

### PR DESCRIPTION
**Pull request by fbianchicodermine**
## Short description
User sees app (PA-PF-PG) in browser's language.

## List of changes proposed in this pull request
- Custom language detector with priority: hash -> sessionStorage -> navigator (browser lang)
- Get current language from i18n.language
- Add listener on i18n event `languageChanged` for update sessionStorage and document attribute `lang` 

## How to test
- Change language in browser settings
- Login as PF, PA, PG

### CODE REVIEW

File: packages/pn-commons/src/components/Footer/__test__/Footer.test.tsx [hunk 1]
```diff
@@ -8,6 +8,17 @@ import Footer from '../Footer';
 
 const mockOpenFn = vi.fn();
 
+const i18n: any = {
```
Comment (SarahDonvito) [inline packages/pn-commons/src/components/Footer/__test__/Footer.test.tsx:11]: Why this type is `any`? Can we use a specific type?

File: packages/pn-personagiuridica-webapp/src/i18n.ts [hunk 2]
```diff
@@ -1,13 +1,24 @@
 import i18next from 'i18next';
 import { initReactI18next } from 'react-i18next';
 import HttpApi from 'i18next-http-backend';
-import { sanitizeString } from '@pagopa-pn/pn-commons';
+import { getLangCode, sanitizeString, setSessionLanguage } from '@pagopa-pn/pn-commons';
+import I18nextBrowserLanguageDetector, { CustomDetector } from 'i18next-browser-languagedetector';
```
Comment (SarahDonvito) [inline packages/pn-personagiuridica-webapp/src/i18n.ts:5]: same as above

File: packages/pn-commons/src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
   companyLegalInfo,
   pagoPALink,
   postLoginLinks,
   preLoginLinks,
 } from '../../utility/costants';
+import { getLangCode } from '../../utility';
+import { useMemo } from 'react';
 
 type Props = {
+  currentLanguage?: string;
```
Comment (SarahDonvito) [inline packages/pn-commons/src/components/Footer/Footer.tsx:13]: Make currentLanguage mandatory

File: packages/pn-pa-webapp/src/i18n.ts [hunk 4]
```diff
@@ -2,23 +2,19 @@ import i18next from 'i18next';
 import LanguageDetector, { CustomDetector } from 'i18next-browser-languagedetector';
 import HttpApi from 'i18next-http-backend';
 import { initReactI18next } from 'react-i18next';
-
-import { sanitizeString, setSessionLanguage } from '@pagopa-pn/pn-commons';
+import { getLangCode, sanitizeString, setSessionLanguage } from '@pagopa-pn/pn-commons';
+import { hashDetectorLookup } from '@pagopa-pn/pn-commons/src/utility/multilanguage.utility';
```
Comment (SarahDonvito) [inline packages/pn-pa-webapp/src/i18n.ts:6]: import hashDetectorLookup  from pn-commons

File: packages/pn-personafisica-webapp/src/i18n.ts [hunk 5]
```diff
@@ -2,12 +2,23 @@ import i18next from 'i18next';
 import { initReactI18next } from 'react-i18next';
 import HttpApi from 'i18next-http-backend';
 import { sanitizeString } from '@pagopa-pn/pn-commons';
+import I18nextBrowserLanguageDetector, { CustomDetector } from 'i18next-browser-languagedetector';
```
[the remark this answers is not in the document]
Comment (fbianchicodermine) [reply packages/pn-personafisica-webapp/src/i18n.ts]: It is an unwanted automatic import.

**Review by alessandrogelmi — APPROVED** [review]

**Pull request merged**